lib/ansible/modules/cloud/hcloud/hcloud_server.py

```diff
@@ -234,10 +234,14 @@ def _create_server(self):
             "server_type": self.client.server_types.get_by_name(
                 self.module.params.get("server_type")
             ),
-            "image": self.client.images.get_by_name(self.module.params.get("image")),
             "user_data": self.module.params.get("user_data"),
             "labels": self.module.params.get("labels"),
         }
+        if self.client.images.get_by_name(self.module.params.get("image")) is not None:
```

Comment:
This could be simplified by

image = self.module.params.get("image")
image_name = self.client.images.get_by_name(image)
image_id = self.client.images.get_by_id(image)

if image_name is not None:
    # When image name is not available look for id instead
    params["image"] = image_name
else: 
    params["image"] = image_id